app/src/main/java/com/example/dictionary/data/datasource/repository/DictionaryRepositoryImpl.kt

```diff
@@ -0,0 +1,52 @@
+package com.example.dictionary.data.datasource.repository
+
+import com.example.dictionary.data.datasource.DictionaryDataSource
+import com.example.dictionary.domain.DictionaryRepository
+import com.example.dictionary.domain.model.ErrorInfo
+import com.example.dictionary.domain.model.WordDetailsModel
+import com.example.dictionary.util.Result
+import javax.inject.Inject
+
+class DictionaryRepositoryImpl @Inject constructor(
+    private val localDS: DictionaryDataSource,
+    private val remoteDS: DictionaryDataSource,
+
+    ) : DictionaryRepository {
+
+    override suspend fun getWordDefinitions(word: String): Result<WordDetailsModel> =
+        when (val response = remoteDS.getWordDefinitions(word)) {
```

Comment:
you should only expose the remote or local call to the VM, for example : override suspend fun getWordDefinitions(word : String) : WordDetailModel = localDs.getWordDefinition(word) ?: remoteDs.getWordDefinition(word), that's it. No need for extra logic, it's not helping at all

Reply:
done